File: convert_heic.py
import os
import glob
from PIL import Image
from pillow_heif import register_heif_opener

register_heif_opener()

def convert_heic_to_jpg(directory):
    heic_files = glob.glob(os.path.join(directory, '*.heic')) + glob.glob(os.path.join(directory, '*.HEIC'))
    print(f"Found {len(heic_files)} HEIC files.")
    
    for file_path in heic_files:
        try:
            print(f"Converting {file_path}...")
            image = Image.open(file_path)
            # Remove alpha channel if present because JPEG doesn't support it
            if image.mode in ("RGBA", "P"):
                image = image.convert("RGB")
            
            jpg_path = os.path.splitext(file_path)[0] + '.jpg'
            image.save(jpg_path, format="JPEG", quality=85)
            
            # Delete original HEIC file
            os.remove(file_path)
        except Exception as e:
            print(f"Failed to convert {file_path}: {e}")

if __name__ == "__main__":
    target_dir = os.path.join(os.getcwd(), 'public', 'images')
    convert_heic_to_jpg(target_dir)
    print("Done!")

File: convert_heic.js
const fs = require('fs');
const path = require('path');
const { promisify } = require('util');
const heicConvert = require('heic-convert');

const readdir = promisify(fs.readdir);
const readFile = promisify(fs.readFile);
const writeFile = promisify(fs.writeFile);
const unlink = promisify(fs.unlink);

async function convertHeicToJpg() {
  const directory = path.join(__dirname, 'public', 'images');
  const files = await readdir(directory);
  const heicFiles = files.filter(f => f.toLowerCase().endsWith('.heic'));
  
  console.log(`Found ${heicFiles.length} HEIC files.`);
  
  for (const file of heicFiles) {
    const filePath = path.join(directory, file);
    const jpgPath = path.join(directory, file.replace(/\.heic$/i, '.jpg'));
    
    try {
      console.log(`Converting ${file}...`);
      const inputBuffer = await readFile(filePath);
      
      const outputBuffer = await heicConvert({
        buffer: inputBuffer, // the HEIC file buffer
        format: 'JPEG',      // output format
        quality: 0.8         // the jpeg compression quality
      });
      
      await writeFile(jpgPath, outputBuffer);
      console.log(`Saved ${jpgPath}`);
      
      // Delete original HEIC file
      await unlink(filePath);
      console.log(`Deleted ${file}`);
    } catch (e) {
      console.error(`Failed to convert ${file}:`, e);
    }
  }
}

convertHeicToJpg().then(() => console.log('Done!'));
